src/components/component/secrets/secret-list-item.tsx

```diff
@@ -28,7 +28,7 @@ export const SecretListItem = ({
 }): JSX.Element => {
   const [isExpanded, setIsExpanded] = useState(false);
 
-  const hasStatusMessages = secret.statusMessages?.length > 0;
+  const hasStatusMessages = secret?.statusMessages?.length > 0;
```

Comment:
There is no point in validating the 'secret' object itself here since it is being used later without being validated in the code (e.g the next line).
It's also technically not an optional object, meaning 'secret' should be validated by the method that called this component and not in here.

```suggestion
  const hasStatusMessages = secret.statusMessages?.length > 0;
```